crates/uv-resolver/src/lock/mod.rs

```diff
@@ -1020,30 +1020,58 @@ impl Lock {
         dist
     }
 
-    /// Return a [`SatisfiesResult`] if the given [`RequiresDist`] does not match the [`Package`].
+    /// Return a [`SatisfiesResult`] if the given extras do not match the [`Package`] metadata.
+    fn satisfies_provides_extra<'lock>(
+        &self,
+        provides_extra: Vec<ExtraName>,
+        package: &'lock Package,
+    ) -> SatisfiesResult<'lock> {
+        if (self.version(), self.revision()) < (1, 1) {
+            return SatisfiesResult::Satisfied;
+        }
+
+        let expected: BTreeSet<_> = provides_extra.iter().collect();
+        let actual: BTreeSet<_> = package.metadata.provides_extras.iter().collect();
+
+        if expected != actual {
+            let expected = provides_extra.into_iter().collect();
+            return SatisfiesResult::MismatchedPackageProvidesExtra(
+                &package.id.name,
+                package.id.version.as_ref(),
+                expected,
+                actual,
+            );
+        }
+
+        SatisfiesResult::Satisfied
+    }
+
+    /// Return a [`SatisfiesResult`] if the given requirements do not match the [`Package`] metadata.
     fn satisfies_requires_dist<'lock>(
-        metadata: RequiresDist,
+        &self,
+        requires_dist: Vec<Requirement>,
+        dependency_groups: BTreeMap<GroupName, Vec<Requirement>>,
         package: &'lock Package,
         root: &Path,
     ) -> Result<SatisfiesResult<'lock>, LockError> {
+        if (self.version(), self.revision()) < (1, 0) {
```

Comment:
(1, 0), not (1, 1)? Oh because it's querying for requires-dist and not provides-extras? (I take it from this there was a v0 lockfile at some point...)

Might want to make this a method with a comprehensible name if it's getting copied so many places. (Or just leave comments for why each version range is chosen)